### ISSUE TITLE
detekt on maven central

### ISSUE THREAD

**Issue by alien11689** [arturbosch/detekt#345]
Hi,
it would be great to have detekt on maven central. Are you planning to release detekt there?

**Comment by arturbosch** [comment]
Not planned yet, but maybe in the future ^^

**Comment by c1phr** [comment]
+1 Another vote for this, my environment unfortunately doesn't allow for using packages that aren't in maven-central.

**Comment by Mauin** [comment]
@arturbosch Looks like detekt is using Bintray for distribution, right? Syncing to MavenCentral should be relatively simple (haven't done it myself yet, though). See: https://bintray.com/docs/usermanual/uploads/uploads_syncingwiththirdpartyplatforms.html

**Comment by arturbosch** [comment]
Opened an issue https://issues.sonatype.org/projects/OSSRH/issues/OSSRH-35016. Should take some days to be able to sync to maven central!

**Comment by c1phr** [comment]
Any update on this? Looks like the Sonatype ticket was resolved.

**Comment by arturbosch** [comment]
Its on my long 'WhenIHaveTime-'TodoList, sorry :/

**Comment by vanniktech** [comment]
Can we pelase do this, @arturbosch?

**Comment by arturbosch** [comment]
I'm on this, syncing with bintray... detekt needs a javadoc.jar ..

**Comment by vanniktech** [comment]
```groovy
project.tasks.create("javadocsJar", Jar.class) {
  classifier = 'javadoc'
  from project.javadoc.destinationDir
}.dependsOn("javadoc")

project.artifacts {
  archives project.javadocsJar
  ...
}
```

**Comment by arturbosch** [comment]
Thanks with kotlin-dsl it was a bit harder but now I have it ^^

**Comment by arturbosch** [comment]
RC7-2 was successfully synced according to bintray ... lets hope :D

**Comment by vanniktech** [comment]
I like it!

**Comment by vanniktech** [comment]
Does not seem like anything on mavenCentral yet - https://repo.maven.apache.org/maven2/io/gitlab/arturbosch/detekt/

**Comment by arturbosch** [comment]
Hmmm bintray says:
```
Sync Status
Last Synced: Fri Jun 01 19:43:29 GMT 2018
Last Sync Status: Successfully synced without closing repo.
Last Sync Errors: N/A
```
I heard that it could also be up to two days.
We will see.

**Comment by arturbosch** [comment]
Is up: https://mvnrepository.com/artifact/io.gitlab.arturbosch.detekt

**Comment by vanniktech** [comment]
I don't seem to be able to get it.

```
* What went wrong:
Could not resolve all files for configuration ':junit4-android-integration-rules:detektCheck'.
> Could not find io.gitlab.arturbosch.detekt:detekt-cli:1.0.0.RC7.
  Searched in the following locations:
      file:/home/niklas/android/sdk/extras/m2repository/io/gitlab/arturbosch/detekt/detekt-cli/1.0.0.RC7/detekt-cli-1.0.0.RC7.pom
      file:/home/niklas/android/sdk/extras/m2repository/io/gitlab/arturbosch/detekt/detekt-cli/1.0.0.RC7/detekt-cli-1.0.0.RC7.jar
      file:/home/niklas/android/sdk/extras/google/m2repository/io/gitlab/arturbosch/detekt/detekt-cli/1.0.0.RC7/detekt-cli-1.0.0.RC7.pom
      file:/home/niklas/android/sdk/extras/google/m2repository/io/gitlab/arturbosch/detekt/detekt-cli/1.0.0.RC7/detekt-cli-1.0.0.RC7.jar
      file:/home/niklas/android/sdk/extras/android/m2repository/io/gitlab/arturbosch/detekt/detekt-cli/1.0.0.RC7/detekt-cli-1.0.0.RC7.pom
      file:/home/niklas/android/sdk/extras/android/m2repository/io/gitlab/arturbosch/detekt/detekt-cli/1.0.0.RC7/detekt-cli-1.0.0.RC7.jar
      https://repo.maven.apache.org/maven2/io/gitlab/arturbosch/detekt/detekt-cli/1.0.0.RC7/detekt-cli-1.0.0.RC7.pom
      https://repo.maven.apache.org/maven2/io/gitlab/arturbosch/detekt/detekt-cli/1.0.0.RC7/detekt-cli-1.0.0.RC7.jar
      https://dl.google.com/dl/android/maven2/io/gitlab/arturbosch/detekt/detekt-cli/1.0.0.RC7/detekt-cli-1.0.0.RC7.pom
      https://dl.google.com/dl/android/maven2/io/gitlab/arturbosch/detekt/detekt-cli/1.0.0.RC7/detekt-cli-1.0.0.RC7.jar
  Required by:
      project :junit4-android-integration-rules

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

Deprecated Gradle features were used in this build, making it incompatible with Gradle 5.0.
See https://docs.gradle.org/4.7/userguide/command_line_interface.html#sec:command_line_warnings

BUILD FAILED in 6s
71 actionable tasks: 1 executed, 70 up-to-date
➜  junit-rules git:(master) ✗ gw build --refresh-dependencies
Parallel execution is an incubating feature.
> Task :junit4-android-integration-rules:detektCheck FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Could not resolve all files for configuration ':junit4-android-integration-rules:detektCheck'.
> Could not find io.gitlab.arturbosch.detekt:detekt-cli:1.0.0.RC7-2.
  Searched in the following locations:
      file:/home/niklas/android/sdk/extras/m2repository/io/gitlab/arturbosch/detekt/detekt-cli/1.0.0.RC7-2/detekt-cli-1.0.0.RC7-2.pom
      file:/home/niklas/android/sdk/extras/m2repository/io/gitlab/arturbosch/detekt/detekt-cli/1.0.0.RC7-2/detekt-cli-1.0.0.RC7-2.jar
      file:/home/niklas/android/sdk/extras/google/m2repository/io/gitlab/arturbosch/detekt/detekt-cli/1.0.0.RC7-2/detekt-cli-1.0.0.RC7-2.pom
      file:/home/niklas/android/sdk/extras/google/m2repository/io/gitlab/arturbosch/detekt/detekt-cli/1.0.0.RC7-2/detekt-cli-1.0.0.RC7-2.jar
      file:/home/niklas/android/sdk/extras/android/m2repository/io/gitlab/arturbosch/detekt/detekt-cli/1.0.0.RC7-2/detekt-cli-1.0.0.RC7-2.pom
      file:/home/niklas/android/sdk/extras/android/m2repository/io/gitlab/arturbosch/detekt/detekt-cli/1.0.0.RC7-2/detekt-cli-1.0.0.RC7-2.jar
      https://repo.maven.apache.org/maven2/io/gitlab/arturbosch/detekt/detekt-cli/1.0.0.RC7-2/detekt-cli-1.0.0.RC7-2.pom
      https://repo.maven.apache.org/maven2/io/gitlab/arturbosch/detekt/detekt-cli/1.0.0.RC7-2/detekt-cli-1.0.0.RC7-2.jar
      https://dl.google.com/dl/android/maven2/io/gitlab/arturbosch/detekt/detekt-cli/1.0.0.RC7-2/detekt-cli-1.0.0.RC7-2.pom
      https://dl.google.com/dl/android/maven2/io/gitlab/arturbosch/detekt/detekt-cli/1.0.0.RC7-2/detekt-cli-1.0.0.RC7-2.jar
  Required by:
      project :junit4-android-integration-rules

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

Deprecated Gradle features were used in this build, making it incompatible with Gradle 5.0.
See https://docs.gradle.org/4.7/userguide/command_line_interface.html#sec:command_line_warnings

BUILD FAILED in 8s
71 actionable tasks: 1 executed, 70 up-to-date
```

Neither RC7 nor RC7-2. My previous link also still shows 404

**Comment by Mauin** [comment]
Looking at mavenCentral the latest releases seem to be available on there. Closing this for now. Feel free to re-open.
